lib/facter/consul.rb

```diff
@@ -0,0 +1,24 @@
+require 'facter'
+
+# Example:
+# $ consul -v
+# Consul v0.5.0
+# Consul Protocol: 2 (Understands back to: 1)
+consul_v = Facter::Util::Resolution.exec("consul -v")
```

Comment:
Def, I can add a test for this.

Also, regarding refactoring, do you think its better to inject those values in a hash or something? So incase in the future we need more than just `version` or `protocol` it would be easily manageable? Or did you mean some other way to refactor? I can imagine adding more consul facts in the future.